public/app/features/scenes/core/SceneObjectBase.tsx

```diff
@@ -198,3 +199,18 @@ export abstract class SceneObjectBase<TState extends SceneObjectState = {}> impl
     return new (this.constructor as any)(clonedState);
   }
 }
+
+/**
+ * This hook is always returning model.state instead of a useState that remembers the last state emitted on the subject
+ * The reason for this is so that if the model instance change this function will always return the latest state.
+ */
+function useSceneObjectState<TState>(model: SceneObjectBase<TState>): TState {
+  const forceUpdate = useForceUpdate();
+
+  useEffect(() => {
+    const s = model.subject.subscribe(forceUpdate);
```

Comment:
I'm not sure I understand this subscription called with `forceUpdate` callback. Do I get it right that when `subject.next` will be called it will result in the `forceUpdate` execution?

Reply:
@dprokop 

yes, basically the same as before where the subject would cause an update to a useState update function. useForceUpdate is 100% the same (has a useState internally). 

The only difference is we do not store the returned state in useState as that is stale when we get a new SceneObject with new observable subject

Reply:
I see, thanks for the explanation @torkelo